cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -158,17 +194,62 @@
   private String configMapName;
   private CoreV1Api coreV1Api;
   private boolean namespaceCreationEnabled;
+  private KubeMasterPathProvider kubeMasterPathProvider;
+  private LocalFileProvider localFileProvider;
+  private Gson gson;
+  private boolean workloadIdentityEnabled;
+  private String workloadIdentityPool;
+  private String workloadIdentityGCPServiceAccountEmail;
+  private String workloadIdentityProvider;
+  private long workloadIdentityServiceAccountTokenTTLSeconds;
+
+  public KubeMasterEnvironment() {
+    gson = new Gson();
```

Comment:
I am wondering why do we initialize it here and not in `initialize` like everything else? Also it looks like it can be final

Reply:
`initialize` is called during runtime but not during unit tests, and this should be initialized regardless of whether `initialize` is called.

Changed the field to `final`.